### PR TITLE
Fix additional warnings

### PR DESCRIPTION
**Pull request by pdalpra**
Those are probably the last warnings I'm confident on fixing without starting to break stuff everywhere :)

### CODE REVIEW

**Comment by typesafe-tools** [comment]
Can one of the admins verify this patch?

**Comment by dwijnand** [comment]
Restarted the mvn-local test that had failed.

Personally I can't review this easily, as I don't understand some of the changes.

I wonder if you could amend the commit message adding detail on anything that isn't pure cosmetics/formatting/parens on why.

**Comment by pdalpra** [comment]
Would commenting the changes in the diff do ?

**Comment by dwijnand** [comment]
Sure, sounds even better.

File: cache/src/test/scala/CacheTest.scala [hunk 1]
```diff
@@ -12,7 +12,7 @@ object CacheTest // extends Properties("Cache test")
   import FileInfo.hash._
   import Ordering._
   import sbinary.DefaultProtocol.FileFormat
-  def test {
+  def test() {
```
Comment (pdalpra) [inline cache/src/test/scala/CacheTest.scala]: Scala doesn't like function returning `Unit` with no params and parens.

File: compile/inc/src/test/scala/sbt/inc/TestCaseGenerators.scala [hunk 2]
```diff
@@ -81,7 +81,7 @@ object TestCaseGenerators {
   private[this] def lzy[T <: AnyRef](x: T) = SafeLazy.strict(x)
 
   def genNameHash(defn: String): Gen[xsbti.api._internalOnly_NameHash] =
-    value(new xsbti.api._internalOnly_NameHash(defn, defn.hashCode()))
+    const(new xsbti.api._internalOnly_NameHash(defn, defn.hashCode()))
```
Comment (pdalpra) [inline compile/inc/src/test/scala/sbt/inc/TestCaseGenerators.scala:84]: `Gen.value` from Scalacheck has been deprecated and replaced by `Gen.const`.

File: main/src/test/scala/ParseKey.scala [hunk 3]
```diff
@@ -123,7 +123,7 @@ object ParseKey extends Properties("Key parser test") {
   // so that it will use the passed-in `Gen` rather than the one imported
   // from TestBuild.
   def structureGenF(f: (Seq[Scope], Env, ProjectRef) => Structure)(implicit mkEnv: Gen[Env]): Gen[Structure] =
-    structureGen((s, e, p) => Gen.value(f(s, e, p)))
+    structureGen((s, e, p) => Gen.const(f(s, e, p)))
```
Comment (pdalpra) [inline main/src/test/scala/ParseKey.scala:126]: `Gen.value` from Scalacheck has been deprecated and replaced by `Gen.const`.

File: main/src/test/scala/TestBuild.scala [hunk 4]
```diff
@@ -214,7 +214,7 @@ object TestBuild {
 
   def genAcyclicDirect[A, T](maxDeps: Gen[Int], keyGen: Gen[T], max: Gen[Int])(make: (T, Seq[A]) => A): Gen[Seq[A]] =
     genAcyclic[A, T](maxDeps, keyGen, max) { t =>
-      Gen.value { deps =>
+      Gen.const { deps =>
```
Comment (pdalpra) [inline main/src/test/scala/TestBuild.scala:217]: `Gen.value` from Scalacheck has been deprecated and replaced by `Gen.const`.

File: run/src/test/scala/sbt/ForkTest.scala [hunk 5]
```diff
@@ -14,9 +14,9 @@ object ForkTest extends Properties("Fork") {
   final val MaximumClasspathLength = 100000
 
   lazy val genOptionName = frequency((9, Some("-cp")), (9, Some("-classpath")), (1, None))
-  lazy val pathElement = listOf1(alphaNumChar).map(_.mkString)
-  lazy val path = listOf1(pathElement).map(_.mkString(File.separator))
-  lazy val genRelClasspath = listOf1(path)
+  lazy val pathElement = nonEmptyListOf(alphaNumChar).map(_.mkString)
+  lazy val path = nonEmptyListOf(pathElement).map(_.mkString(File.separator))
+  lazy val genRelClasspath = nonEmptyListOf(path)
```
Comment (pdalpra) [inline run/src/test/scala/sbt/ForkTest.scala:19]: `Gen.listOf1` from Scalacheck has been deprecated and replaced by `Gen.nonEmptyListOf`.

File: tasks/standard/src/test/scala/Test.scala [hunk 6]
```diff
@@ -12,7 +12,7 @@ object Test extends std.TaskExtra {
   def t3[A, B, C](a: Task[A], b: Task[B], c: Task[C]) = multInputTask[({ type l[L[x]] = (L[A], L[B], L[C]) })#l]((a, b, c))(AList.tuple3)
 
   val a = task(3)
-  val b = task[Boolean](error("test"))
+  val b = task[Boolean](sys.error("test"))
```
Comment (pdalpra) [inline tasks/standard/src/test/scala/Test.scala:15]: `error` from Predef has been deprecated and replaced by `sys.error`.

File: tasks/standard/src/test/scala/checkResult.scala [hunk 7]
```diff
@@ -16,8 +16,8 @@ object checkResult {
           case i: Incomplete =>
             println(i)
             "One or more tasks failed" |: false
-          case e =>
-            e.printStackTrace
+          case e: Throwable =>
+            e.printStackTrace()
```
Comment (pdalpra) [inline tasks/standard/src/test/scala/checkResult.scala:20]: side effecting method.

File: tasks/standard/src/test/scala/checkResult.scala [hunk 8]
```diff
@@ -16,8 +16,8 @@ object checkResult {
           case i: Incomplete =>
             println(i)
             "One or more tasks failed" |: false
-          case e =>
-            e.printStackTrace
+          case e: Throwable =>
```
Comment (pdalpra) [inline tasks/standard/src/test/scala/checkResult.scala:19]: The compiler advises to at least make annotate explicitly the type.

File: util/logic/src/test/scala/sbt/logic/Test.scala [hunk 9]
```diff
@@ -104,7 +104,7 @@ object TestClauses {
     Logic.reduceAll(cs, Set(A, B))
   }
 
-  def all {
+  def all() {
```
Comment (pdalpra) [inline util/logic/src/test/scala/sbt/logic/Test.scala]: Scala doesn't like function returning Unit with no params and parens.

File: compile/interface/src/main/scala/xsbt/ExtractAPI.scala [hunk 10]
```diff
@@ -266,7 +266,7 @@ class ExtractAPI[GlobalType <: CallbackGlobal](val global: GlobalType,
 
   private def mkStructure(info: Type, s: Symbol, inherit: Boolean): xsbti.api.Structure =
     {
-      val (declared, inherited) = info.members.reverse.partition(_.owner == s)
+      val (declared, inherited) = info.members.toList.reverse.partition(_.owner == s)
```
Comment (pdalpra) [inline compile/interface/src/main/scala/xsbt/ExtractAPI.scala:269]: `reverse` has been deprecated and replaced by `toList.reverse`.

File: compile/src/main/scala/sbt/compiler/AnalyzingCompiler.scala [hunk 11]
```diff
@@ -148,7 +148,7 @@ object AnalyzingCompiler {
         } catch { case e: xsbti.CompileFailed => throw new CompileFailed(e.arguments, "Error compiling sbt component '" + id + "'", e.problems) }
         import sbt.Path._
         copy(resources pair rebase(dir, outputDirectory))
-        zip((outputDirectory ***) x_! relativeTo(outputDirectory), targetJar)
+        zip((outputDirectory ***) pair (relativeTo(outputDirectory), false), targetJar)
```
Comment (pdalpra) [inline compile/src/main/scala/sbt/compiler/AnalyzingCompiler.scala:151]: Deprecation

File: ivy/src/main/scala/sbt/ivyint/SbtChainResolver.scala [hunk 12]
```diff
@@ -146,7 +146,7 @@ private[sbt] case class SbtChainResolver(
               Left(ex)
           } finally {
             oldLatest map { _ => doSetLatestStrategy(resolver, oldLatest) }
-            checkInterrupted
+            checkInterrupted()
```
Comment (pdalpra) [inline ivy/src/main/scala/sbt/ivyint/SbtChainResolver.scala:149]: side effecting method

File: ivy/src/main/scala/sbt/ivyint/SbtChainResolver.scala [hunk 13]
```diff
@@ -223,7 +223,7 @@ private[sbt] case class SbtChainResolver(
         case r: FileResource =>
           try {
             val parser = rmr.getDescriptor.getParser
-            val md = parser.parseDescriptor(settings, r.getFile.toURL, r, false)
+            val md = parser.parseDescriptor(settings, r.getFile.toURI.toURL, r, false)
```
Comment (pdalpra) [inline ivy/src/main/scala/sbt/ivyint/SbtChainResolver.scala:226]: Deprecation

File: main/command/src/main/scala/sbt/BasicCommands.scala [hunk 14]
```diff
@@ -134,7 +134,7 @@ object BasicCommands {
   private[this] def className: Parser[String] =
     {
       val base = StringBasic & not('-' ~> any.*, "Class name cannot start with '-'.")
-      def single(s: String) = Completions.single(Completion.displayStrict(s))
+      def single(s: String) = Completions.single(Completion.displayOnly(s))
```
Comment (pdalpra) [inline main/command/src/main/scala/sbt/BasicCommands.scala:137]: deprecation

File: main/command/src/main/scala/sbt/MainLoop.scala [hunk 15]
```diff
@@ -96,7 +96,7 @@ object MainLoop {
     ErrorHandling.wideConvert { state.process(Command.process) } match {
       case Right(s)                  => s
       case Left(t: xsbti.FullReload) => throw t
-      case Left(t)                   => handleException(t, state)
+      case Left(t)                   => state.handleError(t)
```
Comment (pdalpra) [inline main/command/src/main/scala/sbt/MainLoop.scala:99]: deprecation

File: main/command/src/main/scala/sbt/Watched.scala [hunk 16]
```diff
@@ -63,7 +63,7 @@ object Watched {
           case e: Exception =>
             val log = s.log
             log.error("Error occurred obtaining files to watch.  Terminating continuous execution...")
-            MainLoop.handleException(e, s, log)
+            State.handleException(e, s, log)
```
Comment (pdalpra) [inline main/command/src/main/scala/sbt/Watched.scala:66]: deprecation

File: main/src/main/scala/sbt/EvaluateTask.scala [hunk 17]
```diff
@@ -247,7 +247,7 @@ object EvaluateTask {
     {
       val root = ProjectRef(pluginDef.root, Load.getRootProject(pluginDef.units)(pluginDef.root))
       val pluginKey = pluginData
-      val config = extractedConfig(Project.extract(state), pluginDef, state)
+      val config = extractedTaskConfig(Project.extract(state), pluginDef, state)
```
Comment (pdalpra) [inline main/src/main/scala/sbt/EvaluateTask.scala:250]: deprecation

File: main/src/main/scala/sbt/Extracted.scala [hunk 18]
```diff
@@ -40,8 +40,8 @@ final case class Extracted(structure: BuildStructure, session: SessionSettings,
     {
       import EvaluateTask._
       val rkey = resolve(key.scopedKey)
-      val config = extractedConfig(this, structure)
-      val value: Option[(State, Result[T])] = apply(structure, key.task.scopedKey, state, currentRef, config)
+      val config = extractedTaskConfig(this, structure, state)
```
Comment (pdalpra) [inline main/src/main/scala/sbt/Extracted.scala:43]: deprecation

File: main/src/main/scala/sbt/Load.scala [hunk 19]
```diff
@@ -184,7 +184,7 @@ object Load {
   def reapply(newSettings: Seq[Setting[_]], structure: sbt.BuildStructure)(implicit display: Show[ScopedKey[_]]): sbt.BuildStructure =
     {
       val transformed = finalTransforms(newSettings)
-      val newData = makeSettings(transformed, structure.delegates, structure.scopeLocal)
+      val newData = Def.make(transformed)(structure.delegates, structure.scopeLocal, display)
```
Comment (pdalpra) [inline main/src/main/scala/sbt/Load.scala:187]: deprecation

File: main/src/main/scala/sbt/Project.scala [hunk 20]
```diff
@@ -141,7 +141,7 @@ sealed trait Project extends ProjectDefinition[ProjectReference] {
   def aggregate(refs: ProjectReference*): Project = copy(aggregate = (aggregate: Seq[ProjectReference]) ++ refs)
 
   /** Appends settings to the current settings sequence for this project. */
-  def settings(ss: SettingsDefinition*): Project = copy(settings = (settings: Seq[Def.Setting[_]]) ++ ss.flatMap(_.settings))
+  def settings(ss: Def.SettingsDefinition*): Project = copy(settings = (settings: Seq[Def.Setting[_]]) ++ ss.flatMap(_.settings))
```
Comment (pdalpra) [inline main/src/main/scala/sbt/Project.scala:144]: deprecation

File: main/src/main/scala/sbt/Project.scala [hunk 21]
```diff
@@ -464,7 +464,7 @@ object Project extends ProjectExtra {
   def relation(structure: BuildStructure, actual: Boolean)(implicit display: Show[ScopedKey[_]]): Relation[ScopedKey[_], ScopedKey[_]] =
     relation(structure.settings, actual)(structure.delegates, structure.scopeLocal, display)
 
-  private[sbt] def relation(settings: Seq[Setting[_]], actual: Boolean)(implicit delegates: Scope => Seq[Scope], scopeLocal: Def.ScopeLocal, display: Show[ScopedKey[_]]): Relation[ScopedKey[_], ScopedKey[_]] =
+  private[sbt] def relation(settings: Seq[Def.Setting[_]], actual: Boolean)(implicit delegates: Scope => Seq[Scope], scopeLocal: Def.ScopeLocal, display: Show[ScopedKey[_]]): Relation[ScopedKey[_], ScopedKey[_]] =
```
Comment (pdalpra) [inline main/src/main/scala/sbt/Project.scala:467]: deprecation

File: cache/src/test/scala/CacheTest.scala (hunk 1 above)
Comment (eed3si9n) [inline cache/src/test/scala/CacheTest.scala]: Why procedure syntax?

File: util/logic/src/test/scala/sbt/logic/Test.scala (hunk 9 above)
Comment (eed3si9n) [inline util/logic/src/test/scala/sbt/logic/Test.scala]: Procedure style.

**Comment by pdalpra** [comment]
@eed3si9n Indeed, I'll fix those

**Comment by eed3si9n** [comment]
Thanks for tending to the garden that's been weedy.

**Comment by dwijnand** [comment]
Thanks @pdalpra. LGTM.

**Pull request merged**